icu4c/source/tools/toolutil/pkg_genc.cpp

```diff
@@ -847,7 +852,24 @@ getArchitecture(uint16_t *pCPU, uint16_t *pBits, UBool *pIsBigEndian, const char
 #   if defined(_M_IX86)
         *pCPU = IMAGE_FILE_MACHINE_I386;
 #   else
-        *pCPU = IMAGE_FILE_MACHINE_UNKNOWN;
+        // Linker for ClangCL doesn't handle IMAGE_FILE_MACHINE_UNKNOWN the same as
```

Comment:
If I read the existing `U_PLATFORM_HAS_WIN32_API` block correctly it's all about Microsoft's `link.exe`, so it seems odd to now mix in another linker in that. Wouldn't this become easier to read if Microsoft's and LLVM's linkers got each their separate `#if` block?

Reply:
In `platform.h` for `U_PLATFORM_HAS_WIN32_API` it says: `Defines whether the Win32 API is available on the platform.`. This is true for ClangCL, so I've added a new constant `U_CLANG_CL` to distinguish when using Clang on Windows. After that, I changed to using that constant inside the mentioned block.

> Wouldn't this become easier to read if Microsoft's and LLVM's linkers got each their separate #if block

It would, but I do not know how to write linker-specific code (I'm not saying it is impossible). What can be done from my perspective is to detect the compiler (MSVC or Clang) and then infer a linker (link.exe or lld-link.exe), which is more or less what we have now after these changes.